elasticsearch/client.go

```diff
@@ -135,3 +142,65 @@ func newV8Client(apikey, user, pwd string, addresses []string, transport http.Ro
 		Transport: transport,
 	})
 }
+
+type JSONResponse struct {
+	content io.ReadCloser
+	err     error
```

Comment:
what special casing?

Reply:
The `Search` method has a different response type than everything else for no obvious reason. If you added the `status_code` to the `JSONResponse` you could reuse it for the `Search` method.

```
Search(index string, body io.Reader) (int, io.ReadCloser, error)
```

```
type JSONResponse struct {
	content io.ReadCloser
	err     error
}
```

Reply:
The non obvious reason is that with a bigger API usage (introduced in this PR), is better to just use the client transport to avoid code duplication. This wasn't much of a problem when `Search` was introduced in first, and why now `Search` becomes  deprecated.  

imo completely getting rid of it is better than amend it, for reasons given in https://github.com/elastic/apm-server/pull/3063#discussion_r360317852, so I rather don't spend time changing signatures now.